stories/QueryRuleContext.stories.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { connectHits } from 'react-instantsearch-core';
+import {
+  QueryRuleCustomData,
+  QueryRuleContext,
+  Highlight,
+  RefinementList,
+} from 'react-instantsearch-dom';
+import { WrapWithHits } from './util';
+
+type CustomDataItem = {
+  title: string;
+  banner: string;
+  link: string;
+};
+
+type MovieHit = {
+  actors: string[];
+  color: string;
+  genre: string[];
+  image: string;
+  objectID: string;
+  score: number;
+  title: string;
+};
+
+const stories = storiesOf('QueryRuleContext', module);
+
+const StoryHits = connectHits(({ hits }: { hits: MovieHit[] }) => (
+  <div className="hits">
+    {hits.map(hit => (
+      <div key={hit.objectID} className="hit">
+        <div className="hit-picture">
+          <img src={hit.image} />
+        </div>
+
+        <div className="hit-content">
+          <div>
+            <Highlight attribute="title" hit={hit} />
+          </div>
+        </div>
+      </div>
+    ))}
+  </div>
+));
+
+const storyProps = {
+  appId: 'latency',
+  apiKey: 'af044fb0788d6bb15f807e4420592bc5',
+  indexName: 'instant_search_movies',
+  linkedStoryGroup: 'QueryRuleCustomData',
+  hitsElement: <StoryHits />,
+};
+
+stories
+  .add('default', () => (
+    <WrapWithHits {...storyProps}>
+      <ul>
+        <li>
+          On empty query, select the "Drama" category and The Shawshank
+          Redemption appears
+        </li>
+        <li>
+          On empty query, select the "Thriller" category and Pulp Fiction
+          appears
+        </li>
+        <li>
+          Type <q>music</q> and a banner will appear
+        </li>
+      </ul>
+
+      <div style={{ display: 'flex' }}>
+        <aside style={{ flex: 1 }}>
+          <RefinementList attribute="genre" />
+        </aside>
+
+        <main style={{ flex: 2 }}>
+          <QueryRuleContext
+            trackedFilters={{
+              genre: () => ['Drama', 'Thriller'],
+            }}
+          />
+
+          <QueryRuleCustomData>
+            {({ items }: { items: CustomDataItem[] }) => {
+              if (!items[0]) {
+                return null;
+              }
+
+              const { banner, title, link } = items[0];
```

Comment:
in the other implementations (Vue, Angular) of context I mapped over the items. That could be done here too

Reply:
I originally did, but when you select both "Drama" and "Thriller", two banners appear. It might  indeed be better to show everything in the stories.

Reply:
Yes, I found the "two banners" clearer. We might need to do a pass on how to properly display banners though, they take up a _lot_ of space now